esp-hal/src/dma/mod.rs

```diff
@@ -2009,23 +2064,50 @@ impl DmaTxBuf {
     /// Each descriptor can handle 4092 bytes worth of buffer.
     ///
     /// Both the descriptors and buffer must be in DMA-capable memory.
-    /// Only DRAM is supported.
+    /// Only DRAM is supported for descriptors.
     pub fn new(
         descriptors: &'static mut [DmaDescriptor],
         buffer: &'static mut [u8],
     ) -> Result<Self, DmaBufError> {
-        let min_descriptors = buffer.len().div_ceil(CHUNK_SIZE);
+        Self::new_with_chunk_size(descriptors, buffer, CHUNK_SIZE, None)
+    }
+
+    /// Creates a new [DmaTxBuf] from some descriptors and a buffer.
+    ///
+    /// There must be enough descriptors for the provided buffer.
+    /// Each descriptor can handle at most 4092 bytes worth of buffer.
+    /// The chunk size must be greater than 0 and less than or equal to 4092.
+    /// Optionally, a block size can be provided for PSRAM & Burst transfers.
+    ///
+    /// Both the descriptors and buffer must be in DMA-capable memory.
+    /// Only DRAM is supported for descriptors.
+    pub fn new_with_chunk_size(
+        descriptors: &'static mut [DmaDescriptor],
+        buffer: &'static mut [u8],
+        chunk_size: usize,
```

Comment:
Thinking about it now, any reason not to always use the maximum chunk size for the given alignment/block size?

Reply:
So the default for chunk size has always been 4092 to, I believe, deal with alignment for specific peripherals.  If we always choose the max then would we choose 4095. What if it's for a 16 bit transfer - that may cause issues.  If we leave the chunk size parameter, or make it optional and if None choose based on block_size (probably not 4095) and if the block size is None choose the old default.

Reply:
> What if it's for a 16 bit transfer - that may cause issues.

I'm mostly of the opinion that it is the user's job to do the right thing. They could always specify an odd number besides 4095 for the chunk size. I'd prefer if the hal didn't baby sit/hand hold for this.

> I believe, deal with alignment for specific peripherals.

Yes, I2S0 needs the receive descriptors to be word aligned. Though I believe this is an RX only problem iirc.
And that problem can be dealt with once the I2S driver starts using these trait imo.

I think what's really bothering me is:
- The `chunk_size` is being stored in the the buf. Ideally once `descriptor.size()` is set in the constructor, the chunk_size shouldn't be needed anymore. The size field never changes.
- It's a bit awkward having to deal with incompatible chunk_size and block size combinations.
  - What if the chunk_size is less than the block size?
  - What if the chunk_size is not a multiple of the block size?
  - What if the chunk_size is greater than 4095? Sure this could return an error but imo it's simpler if the user just didn't think about it I think.

The story here will be very different for RX of course but we cross that bridge then.

Reply:
ok, I tested 4095 TX on esp32c6 16bit parl_io and esp32s3 16bit i8080 and it looks fine. I'll:
- remove chunk_size as a param and from the struct
- compute the max based on alignment set with none being 4095

Reply:
this means that we need to pass the block size in the descriptor macros.....

Reply:
Yeah about the macros, I'm wondering if it's best to remove the chunk macros and just have macros that give you a DmaTxBuf directly. The user only has to specify theses things once.

Reply:
I'll think about the macros:
- RX and TX chunk sizes can be different
- need more tests

Reply:
So based on [this comment](https://github.com/esp-rs/esp-hal/pull/2179#discussion_r1764933478) We either need chip specific default chunk size or punt for 4092 (word aligned) on all chips.

Reply:
I'm leaning towards doing something chip specific but I have a feeling I'm the only one who thinks it should be this way